platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/collections/LongListDisk.java

```diff
@@ -40,65 +91,128 @@ public class LongListDisk extends LongList {
+            MerkleDbFileUtils.completelyWrite(currentFileChannel, transferBuffer, 0);
+            chunkList.set(firstChunkWithDataIndex, 0L);
+
+            // copy everything except for the first chunk and the last chunk
+            MerkleDbFileUtils.completelyTransferFrom(
+                    currentFileChannel,
+                    fileChannel,
+                    memoryChunkSize,
+                    (long) (totalNumberOfChunks - 2) * memoryChunkSize);
+
+            // copy the last chunk
+            transferBuffer.clear();
+            MerkleDbFileUtils.completelyRead(fileChannel, transferBuffer);
+            transferBuffer.flip();
+            MerkleDbFileUtils.completelyWrite(
+                    currentFileChannel, transferBuffer, (long) (totalNumberOfChunks - 1) * memoryChunkSize);
+
+            for (int i = firstChunkWithDataIndex + 1; i < totalNumberOfChunks; i++) {
+                chunkList.set(i, (long) (i - firstChunkWithDataIndex) * memoryChunkSize);
+            }
+        }
+        // now, once the file is copied, we can create another channel to access it
+        currentFileChannel = FileChannel.open(
+                tempFile, StandardOpenOption.CREATE, StandardOpenOption.READ, StandardOpenOption.WRITE);
+    }
+
+    private static void fillBufferWithZeroes(ByteBuffer transferBuffer) {
+        Arrays.fill(transferBuffer.array(), (byte) IMPERMISSIBLE_VALUE);
+        transferBuffer.clear();
+    }
+
+    private ByteBuffer initOrGetTransferBuffer() {
+        ByteBuffer buffer = TRANSFER_BUFFER_THREAD_LOCAL.get();
+        if (buffer == null) {
+            buffer = ByteBuffer.allocate(memoryChunkSize).order(ByteOrder.nativeOrder());
+            TRANSFER_BUFFER_THREAD_LOCAL.set(buffer);
+        } else {
+            // clean up the buffer
+            buffer.clear();
+        }
+        return buffer;
+    }
+
+    static Path createTempFile(final String sourceFileName) throws IOException {
+        return TemporaryFileBuilder.buildTemporaryDirectory(STORE_POSTFIX).resolve(sourceFileName);
     }
 
     /**
-     * Stores a long at the given index.
-     *
-     * @param index the index to use
-     * @param value the long to store
-     * @throws IndexOutOfBoundsException if the index is negative or beyond the max capacity of the
-     *     list
-     * @throws IllegalArgumentException if the value is zero
+     * Initializes the file channel to a temporary file.
+     * @param path the path to the source file
      */
     @Override
-    public synchronized void put(final long index, final long value) {
-        checkValueAndIndex(value, index);
+    protected void onEmptyOrAbsentSourceFile(final Path path) throws IOException {
+        Path tempFile = createTempFile(path.toFile().getName());
+        currentFileChannel = FileChannel.open(
+                tempFile, StandardOpenOption.CREATE, StandardOpenOption.READ, StandardOpenOption.WRITE);
```

Comment:
As far as I understand, this code path should be almost identical to the constructor for a new long list (not loaded from file). However, this method doesn't initialize `freeChunks`, while the constructor does.

Reply:
There are two execution paths when it comes to the construction:
- `LongListDisk(final int numLongsPerChunk, final long maxLongs, final long reservedBufferLength)`
- `LongListDisk(final Path file, final long reservedBufferLength)`

Both initialize `freeChunks`.   `onEmptyOrAbsentSourceFile` and `readBodyFromFileChannelOnInit` are called from the second constructor